apps/api/pages/api/destination-calendars/[id]/_patch.ts

```diff
@@ -57,13 +60,91 @@ import { schemaQueryIdParseInt } from "~/lib/validations/shared/queryIdTransform
  *        description: Destination calendar not found
  */
 export async function patchHandler(req: NextApiRequest) {
-  const { prisma, query, body } = req;
+  const { userId, isAdmin, prisma, query, body } = req;
   const { id } = schemaQueryIdParseInt.parse(query);
   const parsedBody = schemaDestinationCalendarEditBodyParams.parse(body);
+  let credentialId: number | undefined = undefined;
+  const assignedUserId = isAdmin ? parsedBody.userId || userId : userId;
+
+  let queryConditions = { userId: assignedUserId };
+
+  // when linked with eventTypeId, we need to fetch the userId from the eventTypeId
+  if (parsedBody.eventTypeId) {
+    const eventType = await prisma.eventType.findFirst({
+      where: { id: parsedBody.eventTypeId, userId: assignedUserId },
+    });
+    if (!eventType)
+      throw new HttpError({
+        statusCode: 400,
+        message: "Bad request, eventTypeId invalid",
+      });
+    parsedBody.userId = null;
```

Comment:
Nice catch. 
Yeah, a user linked destination calendar shouldn't like to event type as well. What we can do now because of this restraint is to simply throw error if and when destination calendar is linked to event type but they want to link it to a user, and vice-versa.